src/mol-repr/volume/dot.ts

```diff
@@ -91,26 +97,69 @@ export function VolumeSphereMeshVisual(materialId: number): VolumeVisual<VolumeS
     }, materialId);
 }
 
+const rand = Math.random;
+const offset = Vec3();
+const tmpVec = Vec3();
+function getRandomOffsetFromBasis(a: Vec3, b: Vec3, c: Vec3, cellMax: number, dotSize: number): Vec3 {
+    const scaleFactor = Math.max(0, (cellMax - dotSize) / cellMax);
+    if (scaleFactor <= 0) return Vec3.set(offset, 0, 0, 0);
+
+    const rx = (rand() - 0.5) * scaleFactor;
+    const ry = (rand() - 0.5) * scaleFactor;
+    const rz = (rand() - 0.5) * scaleFactor;
```

Comment:
I think I prefer this. Even with a scaleFactor of cellMax * 0.7 I still see dots lining up (1tqn volume with uniform dot size of 0.1).

Not sure about using the dot size here. I find the perturbed position are a nice effect with bigger dots too.

```suggestion
function getRandomOffsetFromBasis(a: Vec3, b: Vec3, c: Vec3, cellMax: number): Vec3 {
    const rx = (rand() - 0.5) * cellMax;
    const ry = (rand() - 0.5) * cellMax;
    const rz = (rand() - 0.5) * cellMax;
```